fink_utils/sso/spins.py

```diff
@@ -1113,9 +1128,16 @@ def fit_spin(
     # Total RMS, and per-band
     rms = np.sqrt(np.mean(res_lsq.fun**2))
     outdic["rms"] = rms
+    split_at = []
     for filt in ufilters:
         mask = filters == filt
-        outdic["rms_{}".format(filt)] = np.sqrt(np.mean(res_lsq.fun[mask] ** 2))
+        split_at.append(
+            filters[mask].size
+        )  # N. of observations by oredered filter (N_1, N_2, N_3, N_4)
+    res_lsq_byfilter = np.array_split(res_lsq.fun, np.cumsum(split_at))
```

Comment:
Same comment than above -- perhaps easier to maintain if you create a function to split a vector according to a  list of indices:

```python
def some_name(list_of_filters, ordered_vector):
  ufilters = np.unique(filters)
  split_at = []
  for filt in ufilters:
    mask = filters == filt
    split_at.append(filters[mask].size)
  return np.array_split(ordered_vector, np.cumsum(split_at))
```